src/compiler/checker.ts

```diff
@@ -5978,11 +5978,36 @@ module ts {
             return args;
         }
 
+        // In a 'super' call, type arguments are not provided within the CallExpression node itself.
+        // Instead, they must be fetched from the class declaration's base type node.
+        function getEffectiveTypeArguments(callExpression: CallExpression): TypeNode[] {
+            if (callExpression.expression.kind === SyntaxKind.SuperKeyword) {
+                // TODO (drosen): 1) Discuss if checking needs to be done at this point.
+                //                2) Have a test where type arguments are not provided on the base class.
+                //                3) Have a test where the base class is not generic.
+                var containingClass = <ClassDeclaration>getAncestor(callExpression, SyntaxKind.ClassDeclaration);
+                var baseClassTypeNode = getClassBaseTypeNode(containingClass);
+                return baseClassTypeNode.typeArguments;
+            }
+            else {
+                // Ordinary case - simple function invocation.
+                return (<CallExpression>callExpression).typeArguments;
+            }
+        }
+
         function resolveCall(node: CallLikeExpression, signatures: Signature[], candidatesOutArray: Signature[]): Signature {
             var isTaggedTemplate = node.kind === SyntaxKind.TaggedTemplateExpression;
 
-            var typeArguments = isTaggedTemplate ? undefined : (<CallExpression>node).typeArguments;
-            forEach(typeArguments, checkSourceElement);
+            var typeArguments: TypeNode[];
+
+            if (!isTaggedTemplate) {
+                typeArguments = getEffectiveTypeArguments(<CallExpression>node);

```

Comment:
Exactly! It's like if you gave me corn alongside corn-off-the-cob - it's kind of odd to give both.